cmake/modules/extensions.cmake

```diff
@@ -110,6 +110,10 @@ function(zephyr_link_libraries)
   target_link_libraries(zephyr_interface INTERFACE ${ARGV})
 endfunction()
 
+function(zephyr_late_link_libraries)
+  set_property(TARGET zephyr_interface APPEND PROPERTY LATE_LINK_LIBRARIES ${ARGV})
+endfunction()
```

Comment:
a bit unsure on this.

It's one of those functions which solves an immediate problem but once it's available people will start to use it in various places, thus it has a high built-in risk of being used wrongly if people randomly starts to use this function.

Reply:
yes, this function is definitely easy to abuse. I wonder if zephyr_link_libraries should just check for system libraries (-lc -lgcc et al) and do something internally with them?

Reply:
I like the option to handle `-lc` and `-lgcc` (if applicable) internally.

Alternatively, if the toolchain interface were to have some abstraction / function to just name it's late link libraries, then it would be slightly more adaptable.